scripts/build.js

```diff
@@ -1,39 +1,33 @@
 #!/usr/bin/env node
 
-/* eslint-disable no-console */
+const { spawnSync } = require('child_process')
 
-const { exec, spawnSync } = require('child_process')
+const getUpdatedPackageNames = require('./utils/getUpdatedPackageNames')
+const arrayToGlob = require('./utils/arrayToGlob')
 
-exec('./node_modules/.bin/lerna updated --json', (error, stdout) => {
-  if (stdout === '') {
-    console.log('No components have been changed, nothing to do. Exiting.')
-  } else {
-    const updatedPackages = JSON.parse(stdout)
+getUpdatedPackageNames(packageNames => {
+  const scopeGlob = arrayToGlob(packageNames)
 
-    const packageNames = updatedPackages.map(packageObject => packageObject.name)
-    const scopeGlob = packageNames.length === 1 ? packageNames[0] : `{${packageNames.join(',')}}`
-
-    spawnSync(
-      './node_modules/.bin/lerna',
-      [
-        'exec',
-        '--scope',
-        scopeGlob,
-        '--ignore',
-        '@tds/shared-*',
-        '--',
-        '$LERNA_ROOT_PATH/scripts/build.sh',
-      ],
-      {
-        stdio: 'inherit',
-      }
-    )
-    spawnSync(
-      './node_modules/.bin/lerna',
-      ['run', '--scope', scopeGlob, '--ignore', '@tds/shared-*', 'build'],
-      {
-        stdio: 'inherit',
-      }
-    )
-  }
+  spawnSync(
+    './node_modules/.bin/lerna',
+    [
+      'exec',
+      '--scope',
+      scopeGlob,
+      '--ignore',
+      '@tds/shared-*',
+      '--',
+      '$LERNA_ROOT_PATH/scripts/build.sh',
+    ],
+    {
+      stdio: 'inherit',
+    }
+  )
+  spawnSync(
+    './node_modules/.bin/lerna',
+    ['run', '--scope', scopeGlob, '--ignore', '@tds/shared-*', 'build'],
```

Comment:
If you ignore shared components, will those versions not be bumped when there are changes? Or does that not matter since they're private and are always precompiled?

Reply:
I'll check on this...

Reply:
Ahhh, this is just the build script, aka rollup. Shared components don't have their own rollup config, so no version bumping is happening here anyway.

The publish.js script does not ignore anything.